tests/functional_tests/test_groups/models/qwen/test_qwen3_peft_verify.py

```
@@ -160,7 +177,8 @@ def test_cpu_peft_only(self, qwen3_toy_model_dir, adapter_and_checkpoint):
             "--cpu",
         ]
 
-        result = subprocess.run(cmd, capture_output=True, text=True, cwd=PROJECT_ROOT)
+        env = {**os.environ, "MASTER_PORT": str(_find_free_port())}
+        result = subprocess.run(cmd, capture_output=True, text=True, cwd=PROJECT_ROOT, env=env)
```

Ruff (0.15.7)

[error] `subprocess` call: check for execution of untrusted input

(S603)